fred2/sexp_tree.cpp

```diff
@@ -1944,33 +1914,141 @@ SCP_string sexp_tree::match_closest_operator(const SCP_string &str, int node)
 	arg_num = find_argument_number(z, node);
 	opf = query_operator_argument_type(op, arg_num);	// check argument type at this position
 
-	for (i=0; i<(int)Operators.size(); i++) {
-		opr = query_operator_return_type(i);			// figure out which type this operator returns
+	// find the best operator
+	int best = sexp_match_closest_operator(str, opf);
+	if (best < 0)
+		return str;
```

Comment:
This needs to explicitly warn or something.
Also, not sure if just returning str is safe. It's probably _really_ easy to enter something that causes FRED to crash the next time something is done on the SEXP tree

Reply:
I'll add a warning.  But FRED should be able to handle situations with invalid operators now.  I did a lot of stress testing of the sexp_tree code; see #4916 and #4932 for example.